tests/visitable.cpp

```diff
@@ -0,0 +1,21 @@
+#include "catch/catch.hpp"
+
+#include "calendar.h"
+#include "inventory.h"
+#include "item.h"
+
+
+TEST_CASE( "visitable_summation" ) {
+    inventory test_inv;
+
+    item bottle_of_water( "bottle_plastic", calendar::turn );
+    item water_in_bottle( "water", calendar::turn );
+    water_in_bottle.charges = bottle_of_water.get_remaining_capacity_for_liquid( water_in_bottle );
+    bottle_of_water.put_in( water_in_bottle );
+    test_inv.add_item( bottle_of_water );
+
+    item unlimited_water( "water", 0, item::INFINITE_CHARGES );
```

Comment:
The case that broke previously was specific to iteration order (e.g. whether INFINITE_CHARGES was encountered first) and/or having more than one INFINITE_CHARGES item available; it would be good to test those cases